### DISCUSSION TITLE
Using tailwind config customizations

### DISCUSSION THREAD

**Discussion by mohammadizanloo55** [saadeghi/daisyui#2121] [Ideas / feature requests]
## summary
This library is perfect and I thank you for developing it, but we have a  problem with the styles of this library, which is that the spacing of the components or border-radius is hardcoded and cannot be read from tailwind config.

## More details 

for example when my tailwind config file is something like this 
```js
/** @type {import('tailwindcss').Config} */
module.exports = {
  content: [
    "./src/pages/**/*.{js,ts,jsx,tsx,mdx}",
    "./src/components/**/*.{js,ts,jsx,tsx,mdx}",
    "./src/app/**/*.{js,ts,jsx,tsx,mdx}",
  ],
  plugins: [require("daisyui")],
};
```
I have this style in the button element 

```css
padding-left: 1rem;
padding-right: 1rem;
```

when I change the tailwind config file to this 

```js


/** @type {import('tailwindcss').Config} */
module.exports = {
  content: [
    "./src/pages/**/*.{js,ts,jsx,tsx,mdx}",
    "./src/components/**/*.{js,ts,jsx,tsx,mdx}",
    "./src/app/**/*.{js,ts,jsx,tsx,mdx}",
  ],
  theme: {
    spacing: {
      1: "1rem",
      2: "2rem",
      3: "3rem",
      4: "4rem",
    },
  },
  plugins: [require("daisyui")],
};

```

I expect to daisy-ui use my new spacing  and give a style like this in button class

```css
padding-left: 4rem;
padding-right: 4rem;

```

 but I still have the previous style

**Comment by saadeghi** [comment]
We can do that but it would create a lot of unexpected issues.  
- In Tailwind `theme` config, you can customize everything and every value. If someone has a set of custom values for tokens, there is a high chance that it's different from Tailwind's default values. It may have different names, it may have some values removed, etc. So every customization for the tokens will break every component that is using those values under the hood. That's not good.
- I've seen a lot of projects (here on GitHub) that people are using custom tokens (for spacing, colors, font sizes, etc) and they are also using daisyUI. That gives them the freedom to customize those tokens for Tailwind's utility classes without breaking the components.
- daisyUI's components are expected to look just like the examples and demos. For example, buttons have 4 sizes and are expected to look the same when someone use them on their project. Whether they are using a customized set of Tailwind token values or not. Otherwise any component may have unexpected issues because developers just wanted a different set of tokens for utility classes.

So I don't think it would be a good idea for the components to be dependent on the values of utility class tokens.  

Instead of that, you can customize a specific component using utility classes.  
Either on CSS
```css
.btn {
  @apply px-6;
}
```
or in HTML:
```html
class="btn px-6"
```
Or [on daisyUI's theme config](https://daisyui.com/docs/themes/#-8)

This way your customization for the component class names are isolated from your customizations for the utility class names and I think it's important because otherwise there would be no way to customize the tokens without affecting daisyUI components.

Let me know if you have any questions.
